ml-proto/host/builtins.ml

```diff
@@ -1,9 +1,50 @@
 open Source
+open Eval
+open Memory
+open Types
```

Comment:
(Sorry for nit-picking these `open` directives; my experience with OCaml so far has been that having lots of `open`s leads to increased chances of confusion from name collisions or struct type literal inferences). Is this `open` needed?